File: lingting-aws/src/main/java/live/lingting/framework/aws/s3/interfaces/AwsS3Listener.java
package live.lingting.framework.aws.s3.interfaces;

import live.lingting.framework.aws.s3.AwsS3Request;
import live.lingting.framework.http.HttpResponse;
import live.lingting.framework.http.header.HttpHeaders;
import live.lingting.framework.value.multi.StringMultiValue;

import java.time.LocalDateTime;

/**
 * @author lingting 2024/11/5 14:47
 */
public interface AwsS3Listener {

	void onFailed(AwsS3Request request, HttpResponse response);

	void onAuthorization(AwsS3Request request, HttpHeaders headers, StringMultiValue params, LocalDateTime now);

}

File: lingting-aws/src/main/kotlin/live/lingting/framework/aws/s3/interfaces/AwsS3Listener.kt
package live.lingting.framework.aws.s3.interfaces


import live.lingting.framework.aws.AwsSigner
import live.lingting.framework.aws.s3.AwsS3Request
import live.lingting.framework.http.HttpResponse
import live.lingting.framework.http.HttpUrlBuilder
import live.lingting.framework.http.header.HttpHeaders

/**
 * @author lingting 2024/11/5 14:47
 */
interface AwsS3Listener {

    fun onFailed(request: AwsS3Request, response: HttpResponse)

    fun onSign(request: AwsS3Request, headers: HttpHeaders, url: HttpUrlBuilder): AwsSigner<*, *>

}
